src/lib/finalMetrics.ts

```diff
@@ -16,5 +16,6 @@
 
 import {Metric} from '../types.js';
 
-
-export const finalMetrics: WeakSet<Metric> = new WeakSet();
+export const finalMetrics: WeakSet<Metric>|Set<Metric> = 'WeakSet' in window
```

Comment:
Can you update this to be:

```js
export const finalMetrics: WeakSet<Metric> | Set<Metric> =
    typeof 'WeakSet' !== undefined ? new WeakSet() : new Set();
```

For folks using this library on the server for SSR, `window` may not be defined.

Reply:
Changed to `typeof WeakSet` (which will result in either "function"  or "undefined")